.github/workflows/build.yml

```diff
@@ -71,7 +71,7 @@ jobs:
         name: Checkout
         uses: actions/checkout@v2
         with:
-          ref: ${{ github.event.base_ref }}
+          ref: ${{ github.event.repository.master_branch }}
```

Comment:
This is going to prevent releasing off the `release/*` branches which will cause a longer-term issue with back-porting features/fixes to the version that supports the LTS Sonarqube version. Given the priority just now is to release from master so I can get the 8.1 and 8.2 supporting versions out, this isn't a critical issue, but do you have any suggestions for allowing non-master releases?

Reply:
@mc1arke We can take a look at the [Dump context](https://github.com/mc1arke/sonarqube-community-branch-plugin/runs/747411860#step:3:184) to correlate with the information we need. I will make some tests on my fork.

Reply:
@mc1arke From what I could find and confirm is that all the projects I could find tagged their release only from the master branch.

Looking at the best practices, the Git flow is based on [this process](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) for the release branches:

> Once the release is ready to ship, it will get merged it into master and develop, then the release branch will be deleted.

![](https://wac-cdn.atlassian.com/dam/jcr:a9cea7b7-23c3-41a7-a4e0-affa053d9ea7/04%20(1).svg?cdnVersion=1060)

The same applied for this [Git branching model](https://nvie.com/posts/a-successful-git-branching-model/#release-branches):

```
$ git checkout master
Switched to branch 'master'
$ git merge --no-ff release-1.2
Merge made by recursive.
(Summary of changes)
$ git tag -a 1.2
```

Reply:
It wouldn't be practical to try and release this plugin only from master. Since Sonarqube make binary/functional incompatible changes between some of their releases and I'm generally trying to maintain a support the current Sonarqube long-term support version, I need a way of releasing off non-master.

I'm aware this doesn't make releasing particularly simple to manage through an action, so am happy to continue manual releases and have the release job removed from the actions if required.

Reply:
> It wouldn't be practical to try and release this plugin only from master. Since Sonarqube make binary/functional incompatible changes between some of their releases and I'm generally trying to maintain a support the current Sonarqube long-term support version, I need a way of releasing off non-master.

Indeed if you are working on several releases for backwards compatibility, there is no need to create releases only from the master branch.

> I'm aware this doesn't make releasing particularly simple to manage through an action, so am happy to continue manual releases and have the release job removed from the actions if required.

The only problem here is that you have to push a file on the reference branch to the tag at each release. For me this kind of file "should" be immutable and can be summed up to `version=dev`.
But we can always find the reference to the tag and do a proper checkout at the time of release. I'll keep you in touch.

Reply:
@mc1arke Ok I've made some changes do deal with branch ref for a tag.

https://github.com/crazy-max/sonarqube-community-branch-plugin/runs/748014873?check_suite_focus=true

![image](https://user-images.githubusercontent.com/1951866/83981737-2398aa80-a910-11ea-8038-c22d00b3ff2c.png)